dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/AzureOpenAIClientBase.cs

```diff
@@ -30,20 +31,19 @@ public abstract class AzureOpenAIClientBase : ClientBase
         string modelId,
         string endpoint,
         string apiKey,
-        HttpClient? httpClient = null,
+        HttpClient httpClient,
         ILogger? logger = null)
     {
         Verify.NotNullOrWhiteSpace(modelId);
         Verify.NotNullOrWhiteSpace(endpoint);
         Verify.StartsWith(endpoint, "https://", "The Azure OpenAI endpoint must start with 'https://'");
         Verify.NotNullOrWhiteSpace(apiKey);
+        Verify.NotNull(httpClient);
 
         var options = new OpenAIClientOptions();
-
-        if (httpClient != null)
-        {
-            options.Transport = new HttpClientTransport(httpClient);
-        }
+        // The following lines enable the reuse of globally configured SK SDK policies that are configured on the custom HTTPClient.
+        options.Transport = new HttpClientTransport(httpClient);
+        options.RetryPolicy = new NoRetryPolicy();
```

Comment:
What is the default retry policy?

Reply:
There are two mechanisms/properties used by Azure SDK for retry functionality - Retry and RetryPolicy. 

There is not default configuration for the RetryPolicy as far as I can see from source code. 

However, the Retry property assigned a default RetryOptions - https://github.com/Azure/azure-sdk-for-net/blob/1ceb48d8fd964b432a050c6dc024cb76ddbf5cdf/sdk/core/Azure.Core/src/RetryOptions.cs#L12
![image](https://github.com/microsoft/semantic-kernel/assets/68852919/f6cf9c1c-d86a-405f-b2c5-0b5e8a03911c)